topiary-config/src/source.rs

```diff
@@ -21,88 +21,97 @@ impl From<Source> for nickel_lang_core::program::Input<Cursor<String>, OsString>
 }
 
 impl Source {
-    /// Return the valid sources of configuration, in priority order (highest to lowest):
+    /// Iterate through valid sources of configuration, in priority order (highest to lowest):
     ///
     /// 1. `file`, passed as a CLI argument/environment variable
     /// 2. `.topiary/languages.ncl` (or equivalent)
-    /// 3. `~/.config/topiary/languages.ncl` (or equivalent)
-    /// 4. Built-in configuration (per `Self::builtin_nickel()`)
-    pub fn fetch_all(file: &Option<PathBuf>) -> Vec<Self> {
-        let candidates = [
-            ("OS", Some(find_os_configuration_dir_config())),
-            ("workspace", find_workspace_configuration_dir_config()),
+    /// 3. `~/.config/topiary/languages.ncl`
+    /// 4. OS configuration directory (if different from #3)
+    /// 5. Built-in configuration: [`Self::builtin_nickel`]
+    pub fn config_sources(
+        file: &Option<PathBuf>,
+    ) -> impl Iterator<Item = (&'static str, Option<Self>)> {
+        [
             ("CLI", file.clone()),
-        ];
+            ("workspace", find_workspace_configuration_dir_config()),
+            // Certain platforms have alternate config directories (macOS)
+            // polyfill for linux-like `find_os_configuration_dir_config()`
+            // https://docs.rs/directories/latest/src/directories/lib.rs.html#38-43
+            #[cfg(any(
+                target_os = "windows",
+                target_os = "macos",
+                target_os = "ios",
+                target_arch = "wasm32"
+            ))]
+            (
+                "home",
+                std::env::home_dir().map(|home| home.join(".config/topiary/languages.ncl")),
+            ),
+            (
+                "OS",
+                Some(find_os_configuration_dir_config().join("languages.ncl")),
```

Comment:
it was helpful to add the `languages.ncl` early to make `--show-sources` clearer